src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java

```diff
@@ -1034,6 +1040,46 @@ private void renameGitHubRepo(Channel channel, User sender, String repo, String
         }
     }
 
+    /**
+     * Archive a repository and add a prefix to its description (ex: archive explanation)
+     *
+     * @param descriptionPrefix
+     *      Prefix added to the repository description (ex: archive explanation)
+     */
+    private void archiveGitHubRepo(Channel channel, User sender, String repo, String descriptionPrefix) {
+        OutputChannel out = channel.send();
+        try {
+            if (!isSenderAuthorized(channel, sender, false)) {
+                insufficientPermissionError(channel, false);
+                return;
+            }
+            out.message("Archiving " + repo + " and prefixing its description with " + descriptionPrefix);
+
+            GitHub github = GitHub.connect();
+            GHOrganization o = github.getOrganization(IrcBotConfig.GITHUB_ORGANIZATION);
+
+            GHRepository orig = o.getRepository(repo);
+            if (orig == null) {
+                out.message("No such repository: " + repo);
+                return;
+            }
+
+            if (orig.isArchived()) {
+                out.message("The repository " + repo + " is already archived.");
+                return;
+            }
+
+            if (descriptionPrefix != "") {
+                orig.setDescription(descriptionPrefix + orig.getDescription());
```

Comment:
probably needs a space? but might be better to just specify the full description?

Reply:
I was hesitant to put a space here by default, I'll add it.

> might be better to just specify the full description?

I wanted this command to be easy to use, just putting the repo name and a notice, without having to copy and paste the original description.
But it might be better to specify the full description in case this one is outdated/wrong before archiving the repo indeed.